app/api/websocket.py

```
@@ -0,0 +1,316 @@
+        self.active_connections.append(websocket)
+
+        if user_id:
+            if user_id not in self.user_connections:
+                self.user_connections[user_id] = []
+            self.user_connections[user_id].append(websocket)
+
+        logger.info(
+            f"WebSocket connected. Total connections: {len(self.active_connections)}"
+        )
+
+    def disconnect(self, websocket: WebSocket, user_id: str = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)